release/ray_release/retry.py

```diff
@@ -0,0 +1,41 @@
+"""Utils on retry."""
+
+import time
+from functools import wraps
+
+# Default configuration for retry.
+_DEFAULT_MAX_RETRY_COUNT = 10
+_DEFAULT_INIT_DELAY_SEC = 1
+_DEFAULT_MAX_DELAY_SEC = 30
+_DEFAULT_BACKOFF = 2
+_DEFAULT_JITTER_SEC = 1
+_DEFAULT_EXCEPTIONS = (Exception,)
```

Comment:
maybe not set `(Exception,)` as default? maybe use `(TimeoutError,)` or `(IOError,)` ? or force user to provide a non-empty list? or define an `RetryError` in this file and use that as default?

also this is a tuple, so you cannot use this as the default arg value, you need to use `None`, and:

````
if exceptions is None:
   exceptions = ()
````

Reply:
> maybe not set (Exception,) as default? maybe use (TimeoutError,) or (IOError,) ? or force user to provide a non-empty list? 

I think one thing we miss in our error handling is a centralized status code.
One example might be grpc status code: https://grpc.github.io/grpc/core/md_doc_statuscodes.html
which defines which codes are retriable and which are not, also used in tensorflow and abseil.

Here I'm using the `Exception` by default, since I'm mimic `retry.retry`: https://pypi.org/project/retry/
Let me know if you think it acceptable (at least we've specified exception to retry on the release package).